pkg/metrics/metrics.go

```diff
@@ -0,0 +1,63 @@
+package metrics
+
+import "github.com/prometheus/client_golang/prometheus"
+
+const (
+	separator = "_"
+
+	subsystem = "stonesoup_sprayproxy"
+
+	github                    = "github"
+	githubRequestsName        = subsystem + separator + github + separator + "requests_total"
+	forwarded                 = "forwarded"
+	forwardedRequestsName     = subsystem + separator + forwarded + separator + "requests_total"
+	responseTime              = "response" + separator + "time"
+	forwardedResponseTimeName = subsystem + separator + responseTime + separator + "duration_seconds"
+
+	MetricsPort = 6000
+)
+
+var (
+	githubRequestCounter, forwardedRequestCounter = createCounters()
+	responseTimeHistogram                         = createHistogram()
+)
+
+func init() {
+	prometheus.MustRegister(githubRequestCounter)
+	prometheus.MustRegister(forwardedRequestCounter)
+	prometheus.MustRegister(responseTimeHistogram)
+}
```

Comment:
Wondering if it make sense to register the metrics in the `RunServer` function, rather than as an init function. This may make it easier to write the unit tests.

Reply:
On the contrary, I think separating metrics registration from the http server create/start makes unit tests composition easier IMO.  A pattern by the way we established with prior work on build-jenkins openshift stuff.

That said, I think you are on to something wrt using the private built in golang `init()` method alone for metric registration  and complicating unit test composition, since you cannot call the `init()` in metrics.go from `metrics_test.go`.

I'll pivot to performing metric initialization starting from a method callable outside of metrics.go as part of my next push.

Reply:
remember `init()` is special in go. It _always_ runs, even in tests.

Reply:
yes I'm aware 

but the unit tests were previously doing the `init` flow slightly differently ... IMO it is a bit more uniform now